src/main/java/org/sonarlint/intellij/util/SonarLintAppUtils.java

```diff
@@ -53,4 +55,53 @@ public boolean isOpenFile(Project project, VirtualFile file) {
     VirtualFile[] openFiles = FileEditorManager.getInstance(project).getOpenFiles();
     return Arrays.stream(openFiles).anyMatch(f -> f.equals(file));
   }
+
+  /**
+   * If you already know the module where the file belongs, use {@link #getRelativePathForAnalysis(Module, VirtualFile)}
+   * Path will always contain forward slashes.
+   */
+  @CheckForNull
+  public String getRelativePathForAnalysis(Project project, VirtualFile virtualFile) {
```

Comment:
@henryju this file contains the relevant part, rest is just adapting everything to use this.